boot/freeldr/freeldr/arch/i386/pcvideo.c

```diff
@@ -957,6 +971,55 @@ PcVideoGetBufferSize(VOID)
   return ScreenHeight * BytesPerScanLine;
 }
 
+VOID
+PcVideoGetFontsFromFirmware(PULONG RomFontPointers, PULONG ExtendedBIOSDataArea, PULONG ExtendedBIOSDataSize)
+{
+    REGS BiosRegs;
+
+    /* Get the address of the BIOS ROM fonts.
+       Int 10h, AX=1130h, BH = pointer specifier
+       returns: es:bp = address */
+    BiosRegs.d.eax = 0x1130;
+    BiosRegs.b.bh = ROM_8x14CharacterFont;
+    Int386(0x10, &BiosRegs, &BiosRegs);
+    RomFontPointers[0] = BiosRegs.w.es << 4 | BiosRegs.w.bp;
+
+    BiosRegs.b.bh = ROM_8x8DoubleDotFontLo;
+    Int386(0x10, &BiosRegs, &BiosRegs);
+    RomFontPointers[1] = BiosRegs.w.es << 16 | BiosRegs.w.bp;
+
+    BiosRegs.b.bh = ROM_8x8DoubleDotFontHi;
+    Int386(0x10, &BiosRegs, &BiosRegs);
+    RomFontPointers[2] = BiosRegs.w.es << 16 | BiosRegs.w.bp;
+
+    BiosRegs.b.bh = ROM_AlphaAlternate;
+    Int386(0x10, &BiosRegs, &BiosRegs);
+    RomFontPointers[3] = BiosRegs.w.es << 16 | BiosRegs.w.bp;
+
+    BiosRegs.b.bh = ROM_8x16Font;
+    Int386(0x10, &BiosRegs, &BiosRegs);
+    RomFontPointers[4] = BiosRegs.w.es << 16 | BiosRegs.w.bp;
+
+    BiosRegs.b.bh = ROM_Alternate9x16Font;
+    Int386(0x10, &BiosRegs, &BiosRegs);
+    RomFontPointers[5] = BiosRegs.w.es << 16 | BiosRegs.w.bp;
+
+    /* Store address of the extended bios data area in 0x740 */
+    BiosRegs.d.eax = 0xC100;
```

Comment:
Doesn't seem to be directly related to fonts from firmware?

Reply:
Not sure, for video it uses INT 10h, but here INT 15h AH=C1h is used.

http://stanislavs.org/helppc/int_15-c1.html

Reply:
yes I know; what I meant is that looking for the EBDA doesn't seem to have much to do with video fonts.

Reply:
Hence my question above :slightly_smiling_face:  https://github.com/reactos/reactos/pull/1736#issuecomment-512610580

Yeah I'll split it into two functions then.

Reply:
Done.